packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts

```diff
@@ -0,0 +1,793 @@
+import type {
+	IExecuteFunctions,
+	IDataObject,
+	IHttpRequestOptions,
+	IRequestOptionsSimplified,
+	IOAuth2Options,
+	ExecutionError,
+	NodeApiError,
+} from 'n8n-workflow';
+import { NodeConnectionType, NodeOperationError, jsonParse } from 'n8n-workflow';
+
+import set from 'lodash/set';
+import get from 'lodash/get';
+import unset from 'lodash/unset';
+
+import cheerio from 'cheerio';
+import { convert } from 'html-to-text';
+import type {
+	ParameterInputType,
+	ParametersValues,
+	PlaceholderDefinition,
+	ParametersValues as RawParametersValues,
+	SendIn,
+	ToolParameter,
+} from './interfaces';
+
+const getOAuth2AdditionalParameters = (nodeCredentialType: string) => {
+	const oAuth2Options: { [credentialType: string]: IOAuth2Options } = {
+		bitlyOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		boxOAuth2Api: {
+			includeCredentialsOnRefreshOnBody: true,
+		},
+		ciscoWebexOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		clickUpOAuth2Api: {
+			keepBearer: false,
+			tokenType: 'Bearer',
+		},
+		goToWebinarOAuth2Api: {
+			tokenExpiredStatusCode: 403,
+		},
+		hubspotDeveloperApi: {
+			tokenType: 'Bearer',
+			includeCredentialsOnRefreshOnBody: true,
+		},
+		hubspotOAuth2Api: {
+			tokenType: 'Bearer',
+			includeCredentialsOnRefreshOnBody: true,
+		},
+		lineNotifyOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		linkedInOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		mailchimpOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		mauticOAuth2Api: {
+			includeCredentialsOnRefreshOnBody: true,
+		},
+		microsoftDynamicsOAuth2Api: {
+			property: 'id_token',
+		},
+		philipsHueOAuth2Api: {
+			tokenType: 'Bearer',
+		},
+		raindropOAuth2Api: {
+			includeCredentialsOnRefreshOnBody: true,
+		},
+		shopifyOAuth2Api: {
+			tokenType: 'Bearer',
+			keyToIncludeInAccessTokenHeader: 'X-Shopify-Access-Token',
+		},
+		slackOAuth2Api: {
+			tokenType: 'Bearer',
+			property: 'authed_user.access_token',
+		},
+		stravaOAuth2Api: {
+			includeCredentialsOnRefreshOnBody: true,
+		},
+	};
+	return oAuth2Options[nodeCredentialType];
+};
+
+export const configureHttpRequestFunction = async (
+	ctx: IExecuteFunctions,
+	credentialsType: 'predefinedCredentialType' | 'genericCredentialType' | 'none',
+	itemIndex: number,
+) => {
+	if (credentialsType === 'genericCredentialType') {
```

Comment:
Also here, to improve readability, I'd split these `credentialsType` handlers into separate functions